scala-package/infer/src/main/scala/ml/dmlc/mxnet/infer/ImageClassifier.scala

```diff
@@ -0,0 +1,160 @@
+ * limitations under the License.
+ */
+
+package ml.dmlc.mxnet.infer
+
+import ml.dmlc.mxnet._
+
+import scala.collection.mutable.ListBuffer
+
+// scalastyle:off
+import java.awt.image.BufferedImage
+// scalastyle:on
+import java.io.File
+import javax.imageio.ImageIO
+
+
+/**
+  * A class for classifier tasks
+  *
+  * @param modelPathPrefix  PathPrefix from where to load the symbol, parameters and synset.txt
+  *                         Example: file://model-dir/resnet-152(containing resnet-152-symbol.json
+  *                         file://model-dir/synset.txt
+  * @param inputDescriptors Descriptors defining the input node names, shape,
+  *                         layout and Type parameters
+  */
+class ImageClassifier(modelPathPrefix: String,
+                      inputDescriptors: IndexedSeq[DataDesc])
+                      extends Classifier(modelPathPrefix,
+                      inputDescriptors) {
+
+  val classifier: Classifier = getClassifier(modelPathPrefix, inputDescriptors)
+
+  // Loading image from file
+  def loadImageFromFile(inputImagePath: String): BufferedImage = {
+      val img = ImageIO.read(new File(inputImagePath))
+      img
+  }
+
+  /**
+    * Loading input batch of images
+    * @param inputImageDirPath
+    * @return List of buffered images
+    */
+  def loadInputBatch(inputImageDirPath: String): List[BufferedImage] = {
+    val dir = new File(inputImageDirPath)
+    val inputBatch = ListBuffer[BufferedImage]()
+    for (imgFile: File <- dir.listFiles()){
+      val img = ImageIO.read(imgFile)
+      inputBatch += img
+    }
+    inputBatch.toList
+  }
+
+  /**
+    * Reshape the input image to new shape
+    *
+    * @param img       image
+    * @param newWidth  rescale to new width
+    * @param newHeight rescale to new height
+    * @return Rescaled BufferedImage
+    */
+  def getScaledImage(img: BufferedImage, newWidth: Int, newHeight: Int): BufferedImage = {
+    val resizedImage = new BufferedImage(newWidth, newHeight, BufferedImage.TYPE_INT_RGB)
+    val g = resizedImage.createGraphics()
+    g.drawImage(img, 0, 0, newWidth, newHeight, null)
+    g.dispose()
+
+    resizedImage
+  }
+
+  /**
+    * Read image file from provided path
+    *
+    * @param resizedImage BufferedImage to get pixels from
+    * @return NDArray pixels array
+    */
+  def getPixelsFromImage(resizedImage: BufferedImage): NDArray = {
+    val w = resizedImage.getWidth
+    val h = resizedImage.getHeight
+
+    val pixels = new ListBuffer[Float]()
+
+    for (x <- 0 until h) {
+      for (y <- 0 until w) {
+        val color = resizedImage.getRGB(y, x)
+        val red = (color & 0xff0000) >> 16
+        val green = (color & 0xff00) >> 8
+        val blue = color & 0xff
+        pixels += red
+        pixels += green
+        pixels += blue
+      }
+    }
+
+    val reshaped_pixels = NDArray.array(pixels.toArray, shape = Shape(224, 224, 3))
+
+    val swapped_axis = NDArray.swapaxes(reshaped_pixels, 0, 2)
```

Comment:
would you need to swap if you build the Input according to the input descriptor?